odoo/addons/base/models/res_company.py

```diff
@@ -283,6 +283,11 @@ def cache_invalidation_fields(self):
             'sequence', # user._get_company_ids and other potential cached search
         }
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_if_company_has_no_childre(self):
+        if self.child_ids:
```

Comment:
Careful, it can be call in batched.

Reply:
oops, this is embarrassing :sweat_smile: